tools/bundle-build/lib/git.py

```diff
@@ -0,0 +1,25 @@
+import os
+import tempfile
+import subprocess
+
+class GitRepository:
+    def __init__(self, url, ref):
+        self._url = url
+        self._dir = tempfile.TemporaryDirectory()
+        self.execute(f'git init')
+        self.execute(f'git remote add origin {url}')
+        self.execute(f'git fetch --depth 1 origin {ref}')
+        self.execute(f'git checkout FETCH_HEAD')
+        print(f'Checked out {url}@{ref} into {self._dir.name}')
+
+    def dir(self):
+        return self._dir.name
+
+    def execute(self, command):
+        current = os.getcwd()
+        print(f'Executing "{command}" in {self._dir.name}')
+        os.chdir(self._dir.name)
+        try:
+            subprocess.check_call(command, shell = True)
+        finally:
+            os.chdir(current)
```

Comment:
Recommendation: Use the `cwd` argument to `check_call` rather than manually calling `chdir`.